pandas/tests/dtypes/test_cast.py

```diff
@@ -407,3 +408,24 @@ def test_period_dtype(self):
                        np.dtype('datetime64[ns]'), np.object, np.int64]:
             assert find_common_type([dtype, dtype2]) == np.object
             assert find_common_type([dtype2, dtype]) == np.object
+
+    @pytest.mark.parametrize('dtype', [int, float, str, object, None])
+    @pytest.mark.parametrize('datum1', [1, 2., "3", (4, 5), [6, 7], None])
+    @pytest.mark.parametrize('datum2', [8, 9., "10", (11, 12), [13, 14], None])
+    def test_cast_1d_array(self, dtype, datum1, datum2):
+        data = [datum1, datum2]
+        try:
+            # Conversion to 1d array is possible if requested dtype is object
```

Comment:
please separate into 2 tests. This is so confusing as to what cases are what. If you need to skip a few tests on specific cases in the 'good' case, that is ok, but you need to clearly have a separate tests for the error cases. catching errors IN a test is a big no-no. it is impossible to tell what *should* succeed here.